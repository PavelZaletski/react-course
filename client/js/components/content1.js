import React from 'react';

export default () => (
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu lacinia tellus. Sed et quam risus. Phasellus sit amet </p>
);