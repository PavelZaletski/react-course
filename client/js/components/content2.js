import React from 'react';

export default class Content2 extends React.PureComponent {
    render() {
        return (
            <p>Nulla rhoncus tempus erat nec facilisis. Duis quis purus ligula. Lorem ipsum dolor sit amet</p>
        );
    }
}