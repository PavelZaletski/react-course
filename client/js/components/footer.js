import React from 'react';

export default function() {
    return (
        <footer>Netflixroulette</footer>
    );
}