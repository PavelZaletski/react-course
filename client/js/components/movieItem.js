import React from 'react';

export default function ({ item }) {
    const Genres = item.genres.map(genre => <span key={genre}>{genre}</span>);

    return (
        <div className="movie-item">
            <img className="movie-item__img" src={item.poster_path} />
            <div className="movie-item__description">
                <h4 className="movie-item__title">{item.title}</h4>
                <span className="movie-item__release-date">{new Date(item.release_date).getFullYear()}</span>
            </div>
            <div className="movie-item__genres">{Genres}</div>
        </div>
    );
}