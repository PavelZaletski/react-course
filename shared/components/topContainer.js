import React from 'react';

export class TopContainer extends React.Component {
  render() {
    return (
            <div className="top-container" >
                <div className="content-wrapper">
                    <div className="content">
                        <div className="title">Netflixroulette</div>
                        {this.props.children}
                    </div>
                </div>
            </div>
    );
  }
}
