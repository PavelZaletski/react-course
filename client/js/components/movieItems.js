import React from 'react';
import { TopContainer } from './topContainer';
import MovieItem from './movieItem';
import { connect } from 'react-redux';
import { fetchMovieById, fetchMoviesByGenres } from '../actions/movies-actions';

class MovieItems extends React.Component {
    render() {
        const { relatedMovies } = this.props;
        const Items = relatedMovies && relatedMovies.map(item => <MovieItem item={item} key={item.id} />);

        return (
            <div className="movies-list">
                <div className="movies-list__header">
                    <span>Films by {this.props.genre} genre</span>
                    
                </div>
                <div className="movies-list__items">
                    {Items}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (store) => ({
    relatedMovies: store.movies.relatedMovies
});

export default connect(mapStateToProps)(MovieItems);