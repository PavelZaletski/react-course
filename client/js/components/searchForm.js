import React from 'react';
import { TopContainer } from './topContainer';
import { RadioInput } from './radioInput';
import { withRouter } from 'react-router';

class SearchFormComponent extends React.Component {
    state = {
        searchBy: 'title',
        searchText: ''
    };

    onSearchTextChange = (e) => {
        this.setState({
            searchText: e.target.value
        });
    }

    onRadioButtonChange = (searchBy) => {
        this.setState({
            searchBy
        });
    }

    onSubmit = (e) => {
        e.preventDefault();
        const { searchBy, searchText } = this.state;

        this.props.history.push(`/search/${searchBy}/${searchText}`);
    }

    render() {
        const { searchBy } = this.state;
        return (
            <TopContainer>
                <form onSubmit={this.onSubmit}>
                    <div className="search-hint">Find Your movie</div>
                    <input className="search-input" type="text" name="title" required="required" placeholder="Quentin Tarantino" onChange={this.onSearchTextChange}/>

                    <div className="form-footer">
                        <div className="searchby">
                            <span className="filter-text">Search by</span>
                            <RadioInput value="title" name="search" onChange={this.onRadioButtonChange} checked={searchBy === 'title'} />
                            <RadioInput value="genres" name="search" onChange={this.onRadioButtonChange} checked={searchBy === 'genres'} />
                        </div>
                        <input className="search-button btn" type="submit" value="search" required="required" />
                    </div>
                </form>
            </TopContainer>
        );
    }
}

export const SearchForm = withRouter(SearchFormComponent);