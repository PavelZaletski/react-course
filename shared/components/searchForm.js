import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { TopContainer } from './topContainer';
import { RadioInput } from './radioInput';
import { changeSearchBy } from '../actions/movies-actions';

class SearchFormComponent extends React.Component {
    state = {
      searchBy: 'title',
      searchText: '',
    };

    onSearchTextChange = (e) => {
      this.setState({
        searchText: e.target.value,
      });
    }

    onRadioButtonChange = (searchBy) => {
      this.props.changeSearchBy(searchBy);
    }

    onSubmit = (e) => {
      e.preventDefault();
      let { searchText } = this.state;
      searchText = searchText.replace(' ', '-');

      this.props.history.push(`/search/${searchText}`);
    }

    render() {
      const { searchBy } = this.props;
      return (
            <TopContainer>
                <form onSubmit={this.onSubmit}>
                    <div className="search-hint">Find Your movie</div>
                    <input
                      className="search-input"
                      type="text" name="title"
                      required="required"
                      placeholder="Quentin Tarantino"
                      onChange={this.onSearchTextChange}
                    />

                    <div className="form-footer">
                        <div className="searchby">
                            <span className="filter-text">Search by</span>
                            <RadioInput
                              value="title"
                              name="search"
                              onChange={this.onRadioButtonChange}
                              checked={searchBy === 'title'}
                            />
                            <RadioInput
                              value="genres"
                              name="search"
                              onChange={this.onRadioButtonChange}
                              checked={searchBy === 'genres'}
                            />
                        </div>
                        <input className="search-button btn" type="submit" value="search" required="required" />
                    </div>
                </form>
            </TopContainer>
      );
    }
}

const mapStateToProps = store => ({
  searchBy: store.movies.searchBy,
});

const mapDispatchToProps = dispatch => ({
  changeSearchBy: (value) => {
    dispatch(changeSearchBy(value));
  },
});

export const SearchForm = withRouter(connect(mapStateToProps, mapDispatchToProps)(SearchFormComponent));
