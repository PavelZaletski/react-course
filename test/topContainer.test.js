import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import React from 'react';
import { TopContainer } from '../client/js/components/topContainer';
import { SearchForm } from '../client/js/components/searchForm';

describe('TopContainer', () => {
    it('should match snapshot call callback with proper param', () => {
        const eventMock = {
            preventDefault() { }
        };

        const mockCallback = jest.fn();
        let component = shallow(<SearchForm onSearch={mockCallback} />);

        component.find('form').simulate('submit', eventMock);
        expect(mockCallback).toHaveBeenCalled();
        expect(mockCallback).toHaveBeenCalledWith({search: '', searchBy: 'title' });

        component = renderer.create(<SearchForm />).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should call callback with proper param', () => {
        const eventMock = {
            target: {
                value: 'genres'
            }
        };

        const mockCallback = jest.fn();
        let component = mount(<SearchForm onSearch={mockCallback} />);

        component.find("input[value='genres']").simulate('change', eventMock);

        component.find('form').simulate('submit', eventMock);
        expect(mockCallback).toHaveBeenCalled();
        expect(mockCallback).toHaveBeenCalledWith({ search: '', searchBy: 'genres' });
    });

    it('should match snapshot', () => {
        const eventMock = {
            target: {
                value: 'test'
            }
        };

        const mockCallback = jest.fn();
        let component = mount(<SearchForm onSearch={mockCallback} />);

        component.find(".search-input").simulate('change', eventMock);

        component.find('form').simulate('submit', eventMock);
        expect(mockCallback).toHaveBeenCalled();
        expect(mockCallback).toHaveBeenCalledWith({ search: 'test', searchBy: 'title' });
    });
});