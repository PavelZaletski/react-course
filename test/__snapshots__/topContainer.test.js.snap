// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopContainer should match snapshot call callback with proper param 1`] = `
<div
  className="top-container"
>
  <div
    className="content-wrapper"
  >
    <div
      className="content"
    >
      <div
        className="title"
      >
        Netflixroulette
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="search-hint"
        >
          Find Your movie
        </div>
        <input
          className="search-input"
          name="title"
          onChange={[Function]}
          placeholder="Quentin Tarantino"
          required="required"
          type="text"
        />
        <div
          className="form-footer"
        >
          <div
            className="searchby"
          >
            <span
              className="filter-text"
            >
              Search by
            </span>
            <label
              className="checked"
            >
              <input
                name="search"
                onChange={[Function]}
                type="radio"
                value="title"
              />
              <span>
                title
              </span>
            </label>
            <label
              className=""
            >
              <input
                name="search"
                onChange={[Function]}
                type="radio"
                value="genres"
              />
              <span>
                genres
              </span>
            </label>
          </div>
          <input
            className="search-button btn"
            required="required"
            type="submit"
            value="search"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
